preprocessing/nextclade/src/loculus_preprocessing/prepro.py

```diff
@@ -230,6 +268,26 @@ def get_output_metadata(
         spec.args = {} if spec.args is None else spec.args
         input_data: InputMetadata = {}
         input_fields: list[str] = []
+        if output_field == "length":
+            segment_name: SegmentName | None = "main"
```

Comment:
actually we can reuse get_segment(spec, unprocessed) here to get the segment_name